### ISSUE TITLE
`ssi_addr` missing on Linux

### ISSUE THREAD

**Issue by Jake-Shadle** [EmbarkStudios/crash-handling#49]
For some reason, at least on my Linux system, the `ssi_addr` field in the siginfo sent to the signal handler is always 0, even for signals such as `SIGSEGV` where it _should_ be populated with the address. We supposedly? set the correct flags when installing the signal handler https://github.com/EmbarkStudios/crash-handling/blob/e6c580d1984c721936974d89ffd2fe3d9f135d4d/crash-handler/src/linux/state.rs#L234 but...maybe not?

**Comment by Jake-Shadle** [comment]
Ahh interesting, maybe I shot myself in the foot, I changed the address used for testing segfault handling https://github.com/EmbarkStudios/crash-handling/blob/6ff595e0e809ab5028bf76d22c4fb036ed4ffa6e/sadness-generator/src/lib.rs#L122 to make it > `u32::MAX` to ensure that 64-bit addresses were being handled on macos, so maybe I need to tune it more carefully.

**Comment by gabrielesvelto** [comment]
Yes, that's definitely a non-canonical address because it has all the upper 16 bits set. Try `(u32::MAX as u64) + 0x42` instead, or something similar that doesn't fit into 32 bits but doesn't go beyond bit 48.

**Comment by gabrielesvelto** [comment]
FYI rust-minidump now disassembles the crashing instruction and retrieves the address of the actual memory access in this scenario. The exception handler will still return the NULL pointer when hitting non-canonical address but the minidump should have the required information to recreate the real one.
